js/index.js

```diff
@@ -44,10 +44,16 @@ app.controller("IndexController", ["$scope", "$http", 'leafletData', function($s
 
     $scope.currentMarker = marker;
     $("#modal").modal('show');
-    $("#modal").on('click', '#btnClose', function() {
+    $("#modal").on('click', '#btnCancel', function() {
+     marker = new Object();
+    })
+    $("#modal").on('click', '#btnSave', function() {
       $http.post(serverUrl, marker).then(postSuccess, errorMessage);
       marker = {};
     })
+    $('#modal').on('hidden.bs.modal', function () {
+      marker = new Object();
```

Comment:
I would probably extract this into a method named `cancel()` or something because I think there is also a third place we want to call it from: when somebody presses the escape key. Do you think you can handle that too? Is there any way to handle multiple events in one on() method?

Reply:
I will modify that new object comment above but in this case escape is why we need the on hidden event for. I don't think we need any other event handler

Reply:
Oh ok. I tried it locally and using the escape key I still had the same problem as before where the marker wouldn't show up. I don't think it's a big problem though. You should merge this branch! I think it's great.

Reply:
I have merged it. Isn't that how it should be then? Maybe I didn't understood what we wanted to do but I thought we wanted pressing the escape button not to have any effects on the markers as well as the database the same way the cancel button functions. Or is it something else?

Reply:
Ohhhh I see. I thought it was creating the database entries just not showing the marker. You're totally right, I wasn't thinking 😆